app/auth/error/page.tsx

```diff
@@ -0,0 +1,39 @@
+"use client";
+
+import { Fragment, useEffect, useState } from "react";
+
+import { signIn } from "next-auth/react";
+
+/** Custom sign-in error page, see documentation at https://next-auth.js.org/configuration/pages */
+const Page = () => {
```

Comment:
Az ted jsem si vsimnul, ze to mas v Page(). 

1. Neni stastne mit Page jako "use client". Mel byt ten prihlasovaci form/button mit jako komponentu.
2. Ta Page() ma searchParams jako parametr. Takze tam se na to sahat z clienta nemusi.

```
Page({
  searchParams,
}: {
  searchParams: { error?: string };
})
```

Reply:
Super, díky moc, předělám!

Reply:
V čem přesně je nešťastné mít celou stránku (v tomhle případě pouze obsah dialogu) jako `"use client"`? Snažím se obecně posunout tu hranici klientských komponent co nejdál (tj. mít klientských věcí co nejmíň), ale v tomhle případě to znamená třeba extrahovat jedno tlačítko do samostatného souboru, z čehož nejsem nijak nadšený.

Reply:
Rozumim, ja jsem se diky mensimu projektu a WRKS naprogramoval v Next13+ uz docela dost a prave pracovat na urovni malych bloku (cti: tlacitko) pro `use client` neni vubec spatne.

To tlacitko s nicim na te strance neinteraguje. Samo vola jenom funkci, ktera si uz vsecko obstara sama (v tomhle pripade redirect). Prijde mi uplne logicke vytvorit komponentu `SignInTryAgainOfThingsGoWrongButton` a tu udelat `use client`. Prijde mi to stejne samozrejme, jako bych to delal pro formular se zadavanim emailu (a tlacitkem) nebo vyplnovanim XYZ udeju (a tlacitkem).

Je to trochu o posunu mindsetu. Ale podle me to stoji za to, co se udrzitelnosti kodebase tyce. Zacnes pak i premyslet v mensich celcich, pripadne kdyz se to okoreni komplexnejsi logikou, tak zapojenim nejakych kontextu, ktery ty mini-blocky propoji (vs posunuti `use client` o uroven vys).

Reply:
Mně se na tomhle přístupu líbí explicitnější a přesnější vyznačení hranice mezi klientem a serverem a naopak se mně nelíbí, že to má podobu samostatného souboru, takže když chci třeba o kousek posunout hranici, tak to znamená přeorganizovat soubory na disku. Taky mně vadí, že najednou vytečou implementační detaily (třeba nějaké to přihlašovací tlačítko) z jednoho souboru do „globálního scope“, protože to musí být exportované symboly. Ale chápu, předělám to a uvidím. Díky!

Reply:
Tak s moduly naprosto souhlasim, to je v TS svete peklo. Ale programatori si proste musi hlidat scope "odkud co importuju", to je vec, ktera nejde obejit. Pamatuju doby Styled Component, kde pri pokusu o importovani `<W` mi to nabidlo 60 ruznych importu `Wrapper`. 🙈

Nastesti je VSCode doplnovani relativne chytre a chyba importu pri (import all dependencies) se stane vyjimecne.